mungegithub/mungers/triage-filer.go

```diff
@@ -0,0 +1,562 @@
+				for _, buildnum := range job.Builds {
+					row, err := rowMap.rowForBuild(buildnum)
+					if err != nil {
+						return nil, err
+					}
+					if data.Builds.Cols.Started[row] > cutoffTime {
+						buildnumsFilt = append(buildnumsFilt, buildnum)
+					}
+				}
+				//check if any builds failed in the time window
+				if len(buildnumsFilt) > 0 {
+					//create entry for this job in this clust if it doesn't exist
+					if jdata, ok := clust.jobsFailed[job.Name]; !ok {
+						clust.jobsFailed[job.Name] = &JobData{
+							builds:  buildnumsFilt,
+							jobPath: data.Builds.JobPaths[job.Name],
+							rowMap:  rowMap,
+							starts:  data.Builds.Cols.Started,
+						}
+					} else {
+						//combine builds sets
+						for _, build := range buildnumsFilt {
+							contains := false
+							for _, oldBuild := range jdata.builds {
+								if oldBuild == build {
+									contains = true
+									break
+								}
+							}
+							if !contains {
+								jdata.builds = append(jdata.builds, build)
+							}
+						}
+					}
+					//mark that jobID failed the current test
+					clust.testsFailed[test.Name] = append(clust.testsFailed[test.Name], job.Name)
+				}
+			}
+		}
+		//check if there were failures in the time window for this clust
+		if len(clust.testsFailed) > 0 {
+			//clust is valid for the time window, calc failures
+			clust.totalJobs = len(clust.jobsFailed)
+			clust.totalTests = len(clust.testsFailed)
+			clust.totalBuilds = 0
+			for _, jdata := range clust.jobsFailed {
+				clust.totalBuilds += len(jdata.builds)
+			}
+			//record clust
+			clusters = append(clusters, clust)
+		}
+	}
+
+	return clusters, nil
+}
+
+func parseTriageData(jsonIn []byte) (*triageData, error) {
+	var data triageData
+	if err := json.Unmarshal(jsonIn, &data); err != nil {
+		return nil, err
+	}
+
+	//check that important fields are populated
+	if data.Builds.Cols.Started == nil || data.Builds.JobsRaw == nil || data.Builds.JobPaths == nil || data.Clustered == nil {
+		return nil, fmt.Errorf("triage data json is missing required field(s).\n")
+	}
+	//populate 'Jobs' with the BuildToRowMap for each job
+	data.Builds.Jobs = make(map[string]BuildToRowMap)
+	for jobID, mapper := range data.Builds.JobsRaw {
+		switch mapper := mapper.(type) {
+		case []interface{}:
+			//in this case rowMapData is a 3 member array. 0:first buildnum, 1:number of builds 2:start index
+			data.Builds.Jobs[jobID] = ContigRowMap{
+				startBuild: int(mapper[0].(float64)),
+				count:      int(mapper[1].(float64)),
+				startRow:   int(mapper[2].(float64)),
+			}
+		case map[string]interface{}:
+			//in this case rowMapData is a dictionary
+			data.Builds.Jobs[jobID] = DictRowMap(mapper)
+		default:
+			return nil, fmt.Errorf("ERROR: The build number to row index mapping for job '%s' is not an accepted type. Type is: %v", jobID, reflect.TypeOf(mapper))
+		}
+	}
+	return &data, nil
+}
+
+//Cluster represents a cluster of failures identified by the triage page.
+type Cluster struct {
```

Comment:
Why is triageData inappropriate for this? It seems like a few extra methods (like one to discard results >24 hours old) would be sufficient to make it usable. The duplication feels unnecessary.